esp-hal/src/spi/master.rs

```diff
@@ -416,42 +416,178 @@ impl Address {
     }
 }
 
+/// SPI clock source.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[instability::unstable]
+pub enum ClockSource {
+    /// Use the APB clock.
+    Apb,
+    // #[cfg(any(esp32c2, esp32c3, esp32s3))]
+    // Xtal,
+}
+
 /// SPI peripheral configuration
 #[derive(Clone, Copy, Debug, PartialEq, Eq, procmacros::BuilderLite)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 #[non_exhaustive]
 pub struct Config {
-    /// SPI bus clock frequency.
-    pub frequency: HertzU32,
+    /// The saved register value, calculated when first needed.
```

Comment:
it's non-public doc but the comment sent me in a wrong direction when looking at the code - "calculated when first needed" confused me - it's calculated when a corresponding setter is invoked

Reply:
Oops that was old docs